### DISCUSSION TITLE
A way to pass InputEvents to Controls

### DISCUSSION THREAD

**Discussion by EroAxe** [godotengine/godot-proposals#5845] [GUI]
Currently there is no way, at least from what I've seen, to pass InputEvents to other Controls.  Which makes any sort of direct hotkeys etc. akin to Godots Add Node dialog pretty annoying.  Since it seems like they require hardcoding the inputs controlling whatever you want.  Rather than allowing the ability to pass along said InputEvent and just have it handled.

Ex. if you want something to act as if it is focused while not actually being focused.  Like I said akin to Godots Add Node Dialog when it passes the arrow key input to the Tree below it for easier item selection.

I do feel like this could be a useful feature, but I'm currently having a bit of an issue coming up with too many situations, besides some sort of search bar.

**Comment by Sauermann** [comment]
I believe, that you can connect to InputEvents of other Nodes. In the following Screenshot, whenever the `LineEdit`-node receives an InputEvent, the `_on_line_edit_gui_input` function of the `Label`-node is called.
![image](https://user-images.githubusercontent.com/6299227/204078578-6d073d54-9ef5-4e7b-94f4-3710647907f8.png)

**Reply by EroAxe** [reply]
Fair, if you want to connect the signal, what I'm talking about is passing a specific InputEvent to another control.  Akin to what, as I said, Godot does with the Add Node Dialog.  Where they pass the up/down inputs to the search list below the bar.  In the source code they manually call gui_input on it with the given event allowing them to let that control handle it with the default functionality.

Technically you could hand code it doing it based off the InputEvent by passing it to a script on it, but there's no way to just take a Event, and give it to something else, and say "hey use this, it's yours".  Which is what Godot does in the source code.
